### PR TITLE
Fix invoice

### LINKED ISSUE #1216: Order details and invoice
Two things need to be fixed:
1. Invoice's grand total cost equals sum of products' and shipping prices from whole instead of a particular shipment.
2. There should be option to change shipping method when splitting order into two or more shipments. Currently it is not possible; also shipping price is set to 0.

### PR DESCRIPTION
**Pull request by koradon**
Fix total and subtotal price in template.

Fixes #1216

### CODE REVIEW

File: saleor/dashboard/order/utils.py [hunk 1]
```diff
@@ -29,8 +29,12 @@ def _create_pdf(rendered_template, absolute_url):
 def create_invoice_pdf(group_pk, absolute_url):
     group = (DeliveryGroup.objects.prefetch_related(
         'items', 'order', 'order__user', 'order__shipping_address',
```
Comment (Pacu2) [inline saleor/dashboard/order/utils.py]: If you prefetch `some_field__some_fields_field` then `some_field` will be automatically prefetched, no need to add it to the list.

File: templates/dashboard/order/pdf/invoice.html [hunk 2]
```diff
@@ -113,25 +113,37 @@
...
       <td class="right-align">
-        {% gross group.order.get_total_shipping html=True %}
+        {% gross group.shipping_price html=True %}
       </td>
     </tr>
+    {% if first_delivery_group %}
+    {% if group.order.voucher %}
+      <tr>
+        <td>
+          {% trans "Voucher" context "Order voucher header" %} {{ group.order.voucher }}
```
Comment (dominik-zeglen) [inline templates/dashboard/order/pdf/invoice.html]: Wouldn't it be better if we display voucher like 'Voucher (Big Autumn Discount)'?

**Comment by dominik-zeglen** [comment]
When discount is percent-based it should (at least I think it should) be displayed in all invoices, beacuse this type of discount is calculated from sum of all ordered products - including those in other shipment groups. At this moment it works like this:
zrzut ekranu 2017-10-30 o 11 42 06
When you do some math you can see that something is not right.

**Comment by patrys** [comment]
Per-shipping group invoices don't make much sense as you always pay for the entire order. If anything we should fix the scope of invoices.

**Comment by elwoodxblues** [comment]
Okay so we have to bring back generating invoices per order @koradon @dominik-zeglen.

File: saleor/dashboard/order/utils.py [hunk 3]
```diff
@@ -26,14 +26,18 @@ def _create_pdf(rendered_template, absolute_url):
...
 
-def create_invoice_pdf(group_pk, absolute_url):
-    group = (DeliveryGroup.objects.prefetch_related(
-        'items', 'order', 'order__user', 'order__shipping_address',
-        'order__billing_address').get(pk=group_pk))
-    ctx = {'group': group}
+def create_invoice_pdf(order_pk, absolute_url):
+    order = (Order.objects.prefetch_related(
+        'user', 'shipping_address',
+        'billing_address', 'voucher').get(
```
Comment (koradon) [inline saleor/dashboard/order/utils.py]: You need to also prefetch 'groups'.

File: saleor/order/models.py [hunk 4]
```diff
@@ -278,6 +278,10 @@ def get_total(self, **kwargs):
         subtotal = super(DeliveryGroup, self).get_total(**kwargs)
         return subtotal + self.shipping_price
 
+    def get_subtotal_without_voucher(self, **kwargs):
+        subtotal = super(DeliveryGroup, self).get_total(**kwargs)
+        return subtotal
+
```
Comment (koradon) [inline saleor/order/models.py]: This is not needed anymore.

**Comment by elwoodxblues** [comment]
@dominik-zeglen Could you fix those issues with the currency display, that we were looking at today?

File: templates/dashboard/order/pdf/invoice.html [hunk 5]
```diff
@@ -5,39 +5,43 @@
 
 <html>
 <head>
-  <title>Invoice, order ID: {{ group.order.id }}</title>
-  <link rel="stylesheet" type="text/css" href="{% static '/assets/document.css' %}" />
+  <title>Invoice, order ID: {{ order.id }}</title>
```
Comment (elwoodxblues) [inline templates/dashboard/order/pdf/invoice.html]: Missing translation

Comment (patrys) [reply templates/dashboard/order/pdf/invoice.html]: Also "order ID: 3123" does not look like something we'd like users to see.

Comment (koradon) [reply templates/dashboard/order/pdf/invoice.html]: Changed to:
title:
"Invoice for Order #123"
header:
"Order #123"

Comment (elwoodxblues) [reply templates/dashboard/order/pdf/invoice.html]: 👍

**Pull request merged**